src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobBuilder.cs

```diff
@@ -652,24 +651,29 @@ public unsafe void WriteBytes(byte* buffer, int byteCount)
                 Throw.InvalidOperationBuilderAlreadyLinked();
             }
 
-            WriteBytesUnchecked(buffer, byteCount);
+            WriteBytesUnchecked(new ReadOnlySpan<byte>(buffer, byteCount));
         }
 
-        private unsafe void WriteBytesUnchecked(byte* buffer, int byteCount)
+        private void WriteBytesUnchecked(ReadOnlySpan<byte> buffer)
         {
-            int bytesToCurrent = Math.Min(FreeBytes, byteCount);
+            if (buffer.IsEmpty)
+            {
+                return;
+            }
```

Comment:
This wasn't previously being special-cased.  Is it common enough to be worth paying the checking cost for the non-empty cases?

Reply:
Removed.